src/common/types/types.cpp

```diff
@@ -572,7 +644,11 @@ void LogicalType::serialize(Serializer& serializer) const {
     case PhysicalTypeID::ARRAY:
     case PhysicalTypeID::STRUCT:
         extraTypeInfo->serialize(serializer);
+        break;
     default:
+        if (typeID == LogicalTypeID::DECIMAL) {
```

Comment:
I wonder if we can change this to
```
if extraTypeInfo != nullptr
    extraTypeInfo.serialize(serializer)
```

Reply:
`make test` passes when you change the whole switch case to `if extraTypeInfo != nullptr ...`